microprot/scripts/pdb_search.py

```diff
@@ -388,3 +391,156 @@ def report_uncovered_subsequences(hits, query, min_subseq_len=40):
 
     return [u for u in uncovered if (u['sequence'] != '') &
                                     (len(u['sequence']) >= min_subseq_len)]
+
+
+def get_q_id(hit):
+    """ Returns the query ID for a hit.
+
+    Parameters
+    ----------
+    A hit parsed from an HHsearch output file, i.e. dict with at least the key
+    'alignment' which is a dict by itself and comes at least with the key
+    'Q xxx' where xxx is some identifier. The value for this 'Q xxx' key is a
+    third dict which needs to contain the key 'sequence'.
+
+    Returns
+    -------
+    str : The query ID starting with 'Q '.
+    """
+    # find the right ID
+    id_ = [id_ for id_ in hit['alignment'].keys()
```

Comment:
@names: see above.
I changed id_ -> _id
I added notes to the docstring to explain the special scenario:
Notes
    -----
    Each 'hit' has one 'alignment', which comes with different lines. One of
    those lines is 'Q consensus'. Another line is called 'Q xxx' where xxx is
    the ID of the input query sequence. This function find this 'Q xxx' ID.
    We assume that there are only two line names starting with 'Q '.